src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -737,12 +746,17 @@ public void onLoad(FlowExecutionOwner owner) throws IOException {
                 }
             } else {  // See if we can/should resume build
                 if (canResume()) {
+                    // TODO: I think we need to null out persistedClean here and resave the execution so that future
+                    // resumptions are not misled by the fact that persistedClean was true before this resumption.
+                    // CpsThreadGroup.run nulls out persistedClean after every step but it does not save that value
+                    // for PERFORMANCE_OPTIMIZED Pipelines.
```

Comment:
Maybe possible to set a transient field after a restart so that `CpsThreadGroup.run` saves the build even for PERFORMANCE_OPTIMIZED Pipelines? Wouldn't be perfect, but would also address the issue noted in `CpsFlowExecution.checkpoint`.

Reply:
Addressed in https://github.com/jenkinsci/workflow-cps-plugin/pull/363/commits/2214c421c47c01e84a2c7218912f0587799f8252.